request/extractor.go

```diff
@@ -90,7 +90,7 @@ func (e BearerExtractor) ExtractToken(req *http.Request) (string, error) {
 	tokenHeader := req.Header.Get("Authorization")
 	// The usual convention is for "Bearer" to be title-cased. However, there's no
 	// strict rule around this, and it's best to follow the robustness principle here.
-	if tokenHeader == "" || !strings.HasPrefix(strings.ToLower(tokenHeader), "bearer ") {
+	if len(tokenHeader) < 7 || !strings.HasPrefix(strings.ToLower(tokenHeader[:7]), "bearer ") {
```

Comment:
This is late and a nitpick, but `HasPrefix` also slices the front. Is it faster to drop the call completely?

```suggestion
	if len(tokenHeader) < 7 || strings.ToLower(tokenHeader[:7]) != "bearer " {
```

Reply:
We adjusted it in a follow-up PR: https://github.com/golang-jwt/jwt/pull/329/files